alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportToggleMonitorAction.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.transport
+
+import org.opensearch.OpenSearchStatusException
+import org.opensearch.action.support.ActionFilters
+import org.opensearch.action.support.HandledTransportAction
+import org.opensearch.action.support.WriteRequest
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.inject.Inject
+import org.opensearch.commons.alerting.action.AlertingActions
+import org.opensearch.commons.alerting.action.AlertingActions.GET_MONITOR_ACTION_TYPE
+import org.opensearch.commons.alerting.action.AlertingActions.INDEX_MONITOR_ACTION_TYPE
+import org.opensearch.commons.alerting.action.GetMonitorRequest
+import org.opensearch.commons.alerting.action.GetMonitorResponse
+import org.opensearch.commons.alerting.action.IndexMonitorRequest
+import org.opensearch.commons.alerting.action.IndexMonitorResponse
+import org.opensearch.commons.alerting.action.ToggleMonitorRequest
+import org.opensearch.commons.alerting.action.ToggleMonitorResponse
+import org.opensearch.core.action.ActionListener
+import org.opensearch.core.common.io.stream.NamedWriteableRegistry
+import org.opensearch.core.rest.RestStatus
+import org.opensearch.tasks.Task
+import org.opensearch.transport.TransportService
+import java.time.Instant
+
+class TransportToggleMonitorAction @Inject constructor(
+    transportService: TransportService,
+    val client: NodeClient,
+    val namedWriteableRegistry: NamedWriteableRegistry,
+    actionFilters: ActionFilters
+) : HandledTransportAction<ToggleMonitorRequest, ToggleMonitorResponse>(
+    AlertingActions.TOGGLE_MONITOR_ACTION_NAME,
+    transportService,
+    actionFilters,
+    ::ToggleMonitorRequest
+) {
+
+    override fun doExecute(
+        task: Task,
+        request: ToggleMonitorRequest,
+        actionListener: ActionListener<ToggleMonitorResponse>
+    ) {
+        val monitorId = request.monitorId
+        val enabled = request.enabled
+
+        val getMonitorRequest = GetMonitorRequest(
+            monitorId = monitorId,
+            version = -3L,
+            method = request.method,
+            srcContext = null
+        )
+
+        client.execute(
+            GET_MONITOR_ACTION_TYPE,
+            getMonitorRequest,
+            object : ActionListener<GetMonitorResponse> {
+                override fun onResponse(getMonitorResponse: GetMonitorResponse) {
+                    try {
+                        if (getMonitorResponse.monitor == null) {
+                            actionListener.onFailure(
+                                OpenSearchStatusException("Monitor $monitorId not found", RestStatus.NOT_FOUND)
```

Comment:
Small observation. As a best practice, we should end error messages with periods (`.`).